buildconfig/Jenkins/pre-commit

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# Check that the source conforms to the pre-commit framework
+# Args are expected as GIT_SHA1 as $1, PROJECT_SOURCE as $2, JOB_TYPE as $3
+
+if [ $# -ne 3 ]; then
+  echo "Usage: ./pre-commit branch-sha1 project-root job-type"
+  echo "  job_type=master|pull-request  "
+  exit 1
+fi
+
+GIT_SHA=$1
+PROJECT_SOURCE=$2
+JOB_TYPE=$3
```

Comment:
Is this is `JOB_NAME` in other buildscripts?

Reply:
So JOB_TYPE is presently just passed into the script, it's defined by whomever calls the script, it's a kinda bespoke method for this script, so you pass master for your main branch or pull-request for just the diff.

Reply:
Does this mean `buildscript` and `buildscript.bat` need to be modified to call this script?

Reply:
Why would they need to call this script?

Reply:
@peterfpeterson We decided to just have it disabled in the main buildscript using the above option as we'll have separate jobs for them with their own scripts.